inbm/dispatcher-agent/tests/unit/sota/test_snapshot.py

```diff
@@ -55,7 +55,7 @@ def test_ubuntu_snapshot(self, order, rc, err, mock_trtl_single_snapshot, mock_s
     def test_Ubuntu_snapshot_raises1(self, rc, err, mock_pickle_dump, mock_trtl_single_snapshot) -> None:
         factory = SotaOsFactory(
             MockDispatcherBroker.build_mock_dispatcher_broker(), None, []).get_os('Ubuntu')
-        snapshot = factory.create_snapshotter("update", '1', False)
+        snapshot = factory.create_snapshotter("update", '1', False, True)
         with patch('builtins.open', new_callable=mock_open()) as m:
             mock_trtl_single_snapshot.return_value = (rc, err)
             with self.assertRaises(SotaError):
```

Comment:
**Suggestion:** Ensure that the assert_called_once method is called within the context manager to guarantee that it is checked even if an exception is raised. [bug]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            with self.assertRaises(SotaError):
                snapshot.take_snapshot()
            mock_pickle_dump.assert_called_once()
```